pystoned/CNLS.py

```diff
@@ -291,28 +277,23 @@ def get_beta(self):
     def get_residual(self):
         """Return residual value by array"""
         if self.optimization_status == 0:
-            print("Model isn't optimized. Use optimize() method to estimate the model.")
-            return False
+            raise Exception("Model isn't optimized. Use optimize() method to estimate the model.")
         residual = list(self.__model__.epsilon[:].value)
         return np.asarray(residual)
 
     def get_lamda(self):
         """Return beta value by array"""
         if self.optimization_status == 0:
-            print("Model isn't optimized. Use optimize() method to estimate the model.")
-            return False
+            raise Exception("Model isn't optimized. Use optimize() method to estimate the model.")
         if type(self.z) == type(None):
-            # TODO: Replace print by warning
-            print("Without z variable")
-            return
+            raise Exception("Cannot retrieve z variables, since the model is built without z variables.")

```

Comment:
should be:

` raise Exception("Estimated coefficient (lambda) cannot be retrieved due to no z variable included in the model.")`

Reply:
The message has been modified.